### PR TITLE
add github action

### PR DESCRIPTION
**Pull request by ericLemanissier**
It automatically generates and publishes the update page. Note than currently, gnuplot fails with `line -8: all points y value undefined!`. the CSV looks properly generated though. Do you have an idea @BenWiederhake ?
https://github.com/ericLemanissier/serenity-fixmes/actions/runs/3731603500/jobs/6330021113#step:8:18
```
From https://github.com/SerenityOS/serenity
   9feac465dc..f639445456  master     -> origin/master
line -8: all points y value undefined!

Repo has 44462 commits.
Newest commits are: ...[('1ea10bcb738d2541f44dece3a9f6921fa0a10c13', [16](https://github.com/ericLemanissier/serenity-fixmes/actions/runs/3731603500/jobs/6330021113#step:8:17)71447944), ('9feac465dc39fdb284da2ce39dd7e29f133580c2', 1671451669), ('f6394454568d3a43e5b24d9978a93d08d98e19b7', 1671454664)]
(The time is 1671454683, the last commit is 19s ago)
Extended cache by bb491a681dd877d782422e107ee3316e670c4ae8 (now containing 44457 keys) (counting took 2.038033962249756s, saving took 2.86102294921875e-06s)
Extended cache by 12d4bbbd11811659cd8feb759107c3491c0f8d9e (now containing 44458 keys) (counting took 1.2375404834747314s, saving took 1.9073486328125e-06s)
Extended cache by efae6e2270f664dd1e9070d243e92d2229b8e23d (now containing 44459 keys) (counting took 1.28511333465576[17](https://github.com/ericLemanissier/serenity-fixmes/actions/runs/3731603500/jobs/6330021113#step:8:18)s, saving took 2.1457672119140625e-06s)
Extended cache by 1ea10bcb738d2541f44dece3a9f6921fa0a10c13 (now containing 44460 keys) (counting took 1.211791992[18](https://github.com/ericLemanissier/serenity-fixmes/actions/runs/3731603500/jobs/6330021113#step:8:19)75s, saving took 2.6226043701171875e-06s)
Extended cache by 9feac465dc39fdb284da2ce39dd7e29f133580c2 (now containing 44461 keys) (counting took 1.2322025299072266s, saving took 2.384185791015625e-06s)
Extended cache by f6394454568d3a43e5b24d9978a93d08d98e[19](https://github.com/ericLemanissier/serenity-fixmes/actions/runs/3731603500/jobs/6330021113#step:8:20)b7 (now containing 44462 keys) (counting took 1.2297217845916748s, saving took 2.384185791015625e-06s)
Plotting with time_now=1671454692
Traceback (most recent call last):
  File "/home/runner/work/serenity-fixmes/serenity-fixmes/./update_counts.py", line 189, in <module>
    run()
  File "/home/runner/work/serenity-fixmes/serenity-fixmes/./update_counts.py", line 185, in run
    write_graphs(commits_and_dates[-1][1])
  File "/home/runner/work/serenity-fixmes/serenity-fixmes/./update_counts.py", line 141, in write_graphs
    subprocess.run(
  File "/usr/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['gnuplot', '-e', '\n                set terminal png size 1700,900 enhanced;\n                set xdata time;\n                set timefmt "%s";\n                set xlabel "Time";\n                set format x "%Y-%m-%d %H:%M";\n                set ylabel "Fixmes and todos";\n                set datafile separator ",";\n                set output "output_total.png";\n                plot "tagged_history.csv" using 1:2 with lines;\n                \n            set output "output_day.png"; plot [724683492:724769892] "tagged_history.csv" using 1:2 with lines;\n        \n            set output "output_week.png"; plot [724165092:724769892] "tagged_history.csv" using 1:2 with lines;\n        \n            set output "output_month.png"; plot [72[20](https://github.com/ericLemanissier/serenity-fixmes/actions/runs/3731603500/jobs/6330021113#step:8:21)91492:7[24](https://github.com/ericLemanissier/serenity-fixmes/actions/runs/3731603500/jobs/6330021113#step:8:25)769892] "tagged_history.csv" using 1:2 with lines;\n        \n            set output "output_year.png"; plot [69[31](https://github.com/ericLemanissier/serenity-fixmes/actions/runs/3731603500/jobs/6330021113#step:8:32)47492:724769892] "tagged_history.csv" using 1:2 with lines;\n        \n            ']' returned non-zero exit status 1.
Error: Process completed with exit code 1.
```

### CODE REVIEW

**Comment by BenWiederhake** [comment]
I know why it fails, see this comment in the code ;)
https://github.com/BenWiederhake/serenity-fixmes/blob/master/update_counts.py#L23
In short: Set that constant to 0, try it again, then it'll work.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the workflow will run
+on:
+  push:
+    branches: [ "main", "master" ]
+  schedule:
+    - cron:  '*/15 * * * *'
```
Comment (BenWiederhake) [inline .github/workflows/main.yml:10]: Comment: Heh, that's also the schedule I'm currently using.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  workflow_dispatch:
```
Comment (BenWiederhake) [inline .github/workflows/main.yml:11]: Comment: Oh nice! I just read up what this does. Nifty! On the other hand, by simply self-hosting I get this for free :P

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      - uses: actions/checkout@v3
+      - uses: actions/checkout@v3
+        with:
+          ref: "hp-pages"
+          path: "pages"
+      - uses: actions/checkout@v3
+        with:
+          repository: "SerenityOS/serenity"
+          path: "serenity"
+          fetch-depth: "0"
```
Comment (BenWiederhake) [inline .github/workflows/main.yml:28]: fetch-depth 0 doesn't make too much sense. Shouldn't this break printing the full commit log? Also, when checking out older commits later on, the old commits aren't available, thus breaking everything.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.8"
+          cache: 'pip' # caching pip dependencies
+      - run: python3 -m pip install -r requirements.txt
+      - name: Install gnuplot
+        run: sudo apt-get install -y gnuplot-nox
+      - name: Generate web page
+        run: |
+          cp pages/*.json .
```
Comment (BenWiederhake) [inline .github/workflows/main.yml:38]: Copying *all* json files over feels weird. Note that this seems to be necessary in order to preserve `cache.json` across runs. Again, you get this for free simply by not running on a container.

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      - uses: actions/checkout@v3
+        with:
+          repository: "SerenityOS/serenity"
+          path: "serenity"
+          fetch-depth: "0"
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.8"
+          cache: 'pip' # caching pip dependencies
+      - run: python3 -m pip install -r requirements.txt
```
Comment (BenWiederhake) [inline .github/workflows/main.yml:33]: Why set up pip and add a requirements file? That file is empty, so this call does absolutely nothing.

**Review by BenWiederhake — COMMENTED** [review]
Also, I don't intend to move to github action as the thing that (re)builds the website. I like the idea, but I'm honestly happy just not having to worry about setting up the container, because it already runs just fine. Of course, if you're happier running it in the actions container, don't let me stop you :D

EDIT: Why spread these changes across 9 commits?

**Comment by BenWiederhake** [comment]
So since I don't wanna keep up to date with whatever changes the githubactions container have each day, and because it gives me more control, I won't change to the github actions container.

Thank you for the pull though! I learned surprisingly much about github actions thanks to you.

File: .github/workflows/main.yml (hunk 3 above)
Comment (ericLemanissier) [reply .github/workflows/main.yml:28]: according to https://github.com/actions/checkout: `Set fetch-depth: 0 to fetch all history for all branches and tags`. I suppose you interpreted as no history at all ?

Comment (BenWiederhake) [reply .github/workflows/main.yml:28]: Oh! Yeah, I absolutely misunderstood it then. My bad!

File: .github/workflows/main.yml (hunk 5 above)
Comment (ericLemanissier) [reply .github/workflows/main.yml:33]: As you said, It's useless here, just provisions for a future time were some external modules could be needed.

File: .github/workflows/main.yml (hunk 4 above)
Comment (ericLemanissier) [reply .github/workflows/main.yml:38]: ok. The unfortunate thing is that you are the only one having this knowledge.

**Comment by ericLemanissier** [comment]
Thanks for taking the time to answer my PR. I understand your decision, and it's ok.
Just for the record, the advantages of having a github action are:

- trying the project/ contributing to it is a lot easier. If users wants to use github worklows, they get CI for free by forking your repo, and can simply edit any .py file, and get the html page updated as a result. If they do not use github workflows, the workflow file can serve as a documentation of all the steps needed on their CI.
- it is as a safety-net for pull-requests, and helps people understand/correct potential failures in their pull requests.

File: .github/workflows/main.yml (hunk 4 above)
Comment (BenWiederhake) [reply .github/workflows/main.yml:38]: You are the one who wrote that line, so I'm not sure what you mean?

**Pull request closed without merging**